### PR TITLE
Revert `Simplify` to v2.15.1

### PR DESCRIPTION
**Pull request by skarab42**
Revert `Simplify` to v2.15.1 due to breaking change introduced in v2.16.0 #436

### CODE REVIEW

**Comment by skarab42** [comment]
@sindresorhus once merged I will add `SimplifyDeep` and the conditional version.

File: index.d.ts [hunk 1]
```diff
@@ -40,21 +40,21 @@ export {Entry} from './source/entry';
...
+  PositiveInfinity,
+  NegativeInfinity,
+  Finite,
+  Integer,
+  Float,
+  NegativeFloat,
+  Negative,
+  NonNegative,
+  NegativeInteger,
+  NonNegativeInteger,
```
Comment (sindresorhus) [inline index.d.ts:57]: Don't make unrelated changes.

Comment (skarab42) [reply index.d.ts:57]: Sorry, I'm having trouble getting xo to work properly with my setup.

Comment (skarab42) [reply index.d.ts]: @sindresorhus Done, but should `xo --fix` replace spaces with tabs? I have to change my VSCode config every time I work on type-fest (and I often forget). I have installed and configured the XO plugin and deactivated eslint and prettier. What am I doing wrong?

Comment (sindresorhus) [reply index.d.ts]: > but should xo --fix replace spaces with tabs?

Yes, this project uses tab indentation.

File: test-d/simplify.ts [hunk 2]
```diff
@@ -42,45 +42,3 @@ expectType<SomeInterfaceAsTypeWrittenByHand>(valueAsInterface);
...
-		this.id = 'some-class';
-		this.code = 42;
-	}
-
-	someMethod() {
-		return this.code;
-	}
-}
-
-expectType<Simplify<SomeClass>>(new SomeClass());
```
Comment (sindresorhus) [inline test-d/simplify.ts:65]: Let's keep this, but commented out, and with a TODO comment to fix it in the future and with a reference to the issue that caused it to be commented out.

File: index.d.ts (hunk 1 above)
Comment (skarab42) [reply index.d.ts]: > Yes, this project uses tab indentation.

I known, but when I run `xo --fix` in the project it doesn't do the conversion...

**Comment by sindresorhus** [comment]
@skarab42 I noticed we didn't add back the deep ability as `SimplifyDeep`, was that intentional? There's currently no way for users to get back `Simplify<..., {deep: true>`.

**Comment by skarab42** [comment]
@sindresorhus Yes, because obviously the bug occurs as soon as any [condition is introduced](https://github.com/sindresorhus/type-fest/issues/436#issuecomment-1219699899), even a simple `true extends true`, but as I can't reproduce the case I'll have to check with @andyjy. 

I have made an [TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyglgWzAGzgMxAHgCoD4oC8UA3gNoDSEI240cAdlANZUD2aU2AugFycVUakLgF8A3AFgAUNID0szgAs4AZyiqo9VsCgB3VgCcmDAObTQkWIhToQAEQgQwOfETKVqtdYxYh2nXiskVAwHJxwBT2FccWk5BWxFaH0jAEMDVgBXegATdTVgVm9gDJzMgGNoVPoQKHLWXLhgOAbzL3hg2zCwAHVDJnSs3JdCKBLM6AgAD2AIXIKDCagAfhJIoTofNg5uPg6bUMde-sHsnIiPDa4YqD56CAA3CANJGSl5KABab7ipC2hugBGUb7EJYABKEHqBnOKhKpgANLBWAgIAA5Vg5CC4XBiKAfPpGNqWboAJhB1jB3UwkOhsPh9BMSJgKPRmOxuPxCgAolNIOVZnlCnp+mMklAVKlUVBUmoAEYQZCsXQAOjVxIBRwAzBTOocnISBhkzjSoYZ6QZEcjURisTi8QSTsbcnomoo6g0ck0WvRfh9vp9fv8oAAFVgqb0NEMZMBqNxjVhgO6ZBAKl5QZAQNDAZOp55QWJ-dpwABeEGjibjJFdOWAilzabxSTgJkUOc0KbTBdewbDEeaDQAgrl4GXRn3I-QK7GoAAyKxl6cqV7SBizAxoVKVa1srEkaRQKBgdJzdsTgf0Yc5UcQV6H8rKZA5PjEA+Ho8n+hn8OTq83u-dtIIhAA) that illustrates a part of the problem in case you want to report it to the TS team.

**Comment by andyjy** [comment]
> @sindresorhus Yes, because obviously the bug occurs as soon as any [condition is introduced](https://github.com/sindresorhus/type-fest/issues/436#issuecomment-1219699899), even a simple `true extends true`, but as I can't reproduce the case I'll have to check with @andyjy.

It seems reasonable to me to introduce `SimplifyDeep<>` as a separate user-accessible type.  Use of the conditional only caused an error for (my) particularly complex types, I would not expect it to be an issue for simpler cases (which I'd expect to be the majority).

A new user-accessible `SimplifyDeep<>` would enable people to opt-in to utilising it when they specifically wish to - it seems reasonable to expect not to apply it to particularly complex types where it would break. If people do run into issues applying it to overly complex types it would be immediately apparent, and not a breaking change.

(The issue was introducing it within the internals of other types where it was not required and caused existing types to stop working when applied to input types that worked previously.)

**Comment by andyjy** [comment]
> I have made an [TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyglgWzAGzgMxAHgCoD4oC8UA3gNoDSEI240cAdlANZUD2aU2AugFycVUakLgF8A3AFgAUNID0szgAs4AZyiqo9VsCgB3VgCcmDAObTQkWIhToQAEQgQwOfETKVqtdYxYh2nXiskVAwHJxwBT2FccWk5BWxFaH0jAEMDVgBXegATdTVgVm9gDJzMgGNoVPoQKHLWXLhgOAbzL3hg2zCwAHVDJnSs3JdCKBLM6AgAD2AIXIKDCagAfhJIoTofNg5uPg6bUMde-sHsnIiPDa4YqD56CAA3CANJGSl5KABab7ipC2hugBGUb7EJYABKEHqBnOKhKpgANLBWAgIAA5Vg5CC4XBiKAfPpGNqWboAJhB1jB3UwkOhsPh9BMSJgKPRmOxuPxCgAolNIOVZnlCnp+mMklAVKlUVBUmoAEYQZCsXQAOjVxIBRwAzBTOocnISBhkzjSoYZ6QZEcjURisTi8QSTsbcnomoo6g0ck0WvRfh9vp9fv8oAAFVgqb0NEMZMBqNxjVhgO6ZBAKl5QZAQNDAZOp55QWJ-dpwABeEGjibjJFdOWAilzabxSTgJkUOc0KbTBdewbDEeaDQAgrl4GXRn3I-QK7GoAAyKxl6cqV7SBizAxoVKVa1srEkaRQKBgdJzdsTgf0Yc5UcQV6H8rKZA5PjEA+Ho8n+hn8OTq83u-dtIIhAA) that illustrates a part of the problem in case you want to report it to the TS team.

I saw someone posted the following explanation in [this StackOverflow answer](https://stackoverflow.com/a/57683652) - it seems to be TypeScript working as designed.

> The conditional type inference is conceptually a no-op, but it's used to [distribute union types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types) and to force the compiler to evaluate the "true" branch of the conditional (if you redefine Expand<T> without it, sometimes the compiler will just output the mapped type {[K in keyof RelevantType]: RelevantType[K]}, which is not what you want to see).
